packages/ethereum-contracts/contracts/superfluid/FlowNFTBase.sol

```diff
@@ -0,0 +1,389 @@
+    }
+
+    /// @inheritdoc IERC721
+    function ownerOf(
+        uint256 tokenId
+    ) public view virtual override returns (address) {
+        address owner = _ownerOf(tokenId);
+        if (owner == address(0)) {
+            revert CFA_NFT_INVALID_TOKEN_ID();
+        }
+        return owner;
+    }
+
+    /// @notice Returns a hardcoded balance of 1
+    /// @dev We always return 1 to avoid the need for additional mapping
+    /// @return balance = 1
+    function balanceOf(
+        address // owner
+    ) external pure returns (uint256 balance) {
+        balance = 1;
+    }
+
+    /// @notice Returns the name of the NFT
+    /// @dev Should follow the naming convention: TOKENx Constant Outflow/Inflow NFT
+    /// @return name of the NFT
+    function name() external view virtual override returns (string memory) {
+        return _name;
+    }
+
+    /// @notice Returns the symbol of the NFT
+    /// @dev Should follow the naming convention: TOKENx(COF/CIF)
+    /// @return symbol of the NFT
+    function symbol() external view virtual override returns (string memory) {
+        return _symbol;
+    }
+
+    /// @notice This returns the Uniform Resource Identifier (URI), where the metadata for the NFT lives.
+    /// @dev Returns the Uniform Resource Identifier (URI) for `tokenId` token.
+    /// @return the token URI
+    function tokenURI(
+        uint256 tokenId
+    ) external view virtual override returns (string memory) {
+        FlowNFTData memory flowData = flowDataByTokenId(tokenId);
+        address superTokenAddress = address(superToken);
+
+        string memory superTokenSymbol = superToken.symbol();
+
+        (
+            uint256 startDate,
+            int96 flowRate, ,
+        ) = CONSTANT_FLOW_AGREEMENT_V1.getFlow(superToken, flowData.flowSender, flowData.flowReceiver);
+
+        return
+            string(
+                abi.encodePacked(
+                    BASE_URI,
+                    "?chain_id=",
+                    block.chainid.toString(),
+                    "&token_address=",
+                    Strings.toHexString(
+                        uint256(uint160(superTokenAddress)),
+                        20
+                    ),
+                    "&token_symbol=",
+                    superTokenSymbol,
+                    "&token_decimals=",
+                    uint256(18).toString(),
+                    "&sender=",
+                    Strings.toHexString(
+                        uint256(uint160(flowData.flowSender)),
+                        20
+                    ),
+                    "&receiver=",
+                    Strings.toHexString(
+                        uint256(uint160(flowData.flowReceiver)),
+                        20
+                    ),
+                    "&flowRate=",
+                    uint256(uint96(flowRate)).toString(),
+                    "&start_date=",
+                    startDate.toString()
```

Comment:
@d10r is it supposed to be that the date of the update goes here for the `start_date`?